rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -242,11 +252,57 @@ class PlayerImpl
     std::shared_ptr<rclcpp::GenericPublisher> publisher_;
     std::function<void(const rclcpp::SerializedMessage &)> publish_func_;
   };
+
+  class PlayerClient final
+  {
+public:
+    explicit PlayerClient(
+      std::shared_ptr<rclcpp::GenericClient> cli,
+      std::string service_name,
+      std::string service_event_type,
+      const rclcpp::Logger & logger);
+
+    // Can call this function if check_include_request_message() return true
+    void async_send_request(const rclcpp::SerializedMessage & message);
+
+    std::shared_ptr<rclcpp::GenericClient> generic_client()
+    {
+      return client_;
+    }
+
+    // Check if message can be unpacked to get request message
+    bool is_include_request_message(const rclcpp::SerializedMessage & message);
+
+private:
+    std::shared_ptr<rclcpp::GenericClient> client_;
+    std::string service_name_;
+    const rclcpp::Logger & logger_;
+    enum class introspection_type
+    {
+      UNKNOW,
```

Comment:
Comments from MichaelOrlov

```suggestion
      UNKNOWN = 0,
```

Fix for typo

Reply:
Updated